### PR TITLE
Logicalcluster pkg move

### PR DESCRIPTION
**Pull request by fabianvf**
Update all references to kcp-dev/apimachinery/pkg/logicalcluster to point ot kcp-dev/logicalcluster

related to kcp-dev/kcp#940

### CODE REVIEW

**Comment by ncdc** [comment]
To add a dependency & get everything updated "the kube way":
1. `hack/pin-dependency.sh github.com/kcp-dev/logicalcluster <ref>`
2. `hack/update-vendor.sh`

**Comment by ncdc** [comment]
Make sure the kcp-dev/apimachinery package isn't referenced in go.mod/code anywhere after this

**Comment by fabianvf** [comment]
>   hack/pin-dependency.sh github.com/kcp-dev/logicalcluster <ref>
>   hack/update-vendor.sh

Thank you, I could tell I hadn't done it right but also had not yet found where the right way was documented

**Comment by fabianvf** [comment]
> > hack/pin-dependency.sh github.com/kcp-dev/logicalcluster 
> > hack/update-vendor.sh
> 
> Thank you, I could tell I hadn't done it right but also had not yet found where the right way was documented

Oh it's documented in the go.mod itself :facepalm: never thought to look there

**Comment by ncdc** [comment]
Want to wait to merge this until we have a corresponding kcp PR that adapts to these changes so we can merge them roughly together

File: go.sum [hunk 1]
```diff
@@ -595,7 +598,7 @@ k8s.io/klog/v2 v2.30.0 h1:bUO6drIvCIsvZ/XFgfxoGFQU/a4Qkh0iAlvUR7vlHJw=
 k8s.io/klog/v2 v2.30.0/go.mod h1:y1WjHnz7Dj687irZUWR/WLkLc5N1YHtjLdmgWjndZn0=
 k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 h1:E3J9oCLlaobFUqsjG9DfKbP2BmgwBL2p7pn0A3dG9W4=
 k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65/go.mod h1:sX9MT8g7NVZM5lVL/j8QyCCJe8YSMW30QvGZWaCIDIk=
-k8s.io/kubernetes v1.23.5/go.mod h1:avI3LUTUYZugxwh52KMVM7v9ZjB5gYJ6D3FIoZ1SHUo=
+k8s.io/kubernetes v1.24.0/go.mod h1:8e8maMiZzBR2/8Po5Uulx+MXZUYJuN3vtKwD4Ct1Xi0=
```
Comment (fabianvf) [inline go.sum]: hang on this actually seems bad

File: go.mod [hunk 2]
```diff
@@ -503,6 +504,7 @@ replace (
 	k8s.io/kube-scheduler => ./staging/src/k8s.io/kube-scheduler
 	k8s.io/kubectl => ./staging/src/k8s.io/kubectl
 	k8s.io/kubelet => ./staging/src/k8s.io/kubelet
+	k8s.io/kubernetes => k8s.io/kubernetes v1.23.5
```
Comment (fabianvf) [inline go.mod]: this stopped the other versions from getting bumped, I think the issue was that the apiextensions-apiserver imports kubernetes and the vendor script auto bumps that. I imagine this is not what we want to do but getting closer at least

**Comment by ncdc** [comment]
LGTM, will merge when https://github.com/kcp-dev/kcp/pull/1002 with this branch is green

**Comment by ncdc** [comment]
https://github.com/kcp-dev/kcp/pull/1002 is green, merging!

**Pull request merged**